x-pack/plugin/ml/qa/multi-cluster-tests-with-security/src/test/resources/rest-api-spec/test/multi_cluster/50_sparse_vector.yml

```diff
@@ -184,3 +198,346 @@ teardown:
+                index_options:
+                  prune: false
+                  pruning_config:
+                    tokens_freq_ratio_threshold: 1.0
+                    tokens_weight_threshold: 0.4
+              embeddings:
+                type: sparse_vector
+
+  - match: { status: 400 }
+  - match: { error.type: "mapper_parsing_exception" }
+  - match: { error.reason: "Failed to parse mapping: [index_options] field [pruning_config] should not be set if [prune] is false" }
+
+---
+"Check sparse_vector token pruning index_options tokens freq out of bounds":
+
+  - requires:
+      cluster_features: 'sparse_vector_index_options_supported'
+      reason: "sparse_vector token pruning index options added support in 9.1"
+  - skip:
+      features: headers
+
+  - do:
+      catch: bad_request
+      headers:
+        Authorization: "Basic dGVzdF91c2VyOngtcGFjay10ZXN0LXBhc3N3b3Jk" #test_user credentials
+        Content-Type: application/json
+      indices.create:
+        index: sparse_vector_pruning_test
+        body:
+          mappings:
+            properties:
+              text:
+                type: text
+              ml.tokens:
+                type: sparse_vector
+                index_options:
+                  prune: true
+                  pruning_config:
+                    tokens_freq_ratio_threshold: 101.0
+                    tokens_weight_threshold: 0.4
+              embeddings:
+                type: sparse_vector
+
+  - match: { status: 400 }
+  - match: { error.type: "mapper_parsing_exception" }
+  - match: { error.reason: "Failed to parse mapping: [pruning_config] field [tokens_freq_ratio_threshold] field should be a number between 1 and 100" }
+
+---
+"Check sparse_vector token pruning index_options tokens weight out of bounds":
+
+  - requires:
+      cluster_features: 'sparse_vector_index_options_supported'
+      reason: "sparse_vector token pruning index options added support in 9.1"
+  - skip:
+      features: headers
+
+  - do:
+      catch: bad_request
+      headers:
+        Authorization: "Basic dGVzdF91c2VyOngtcGFjay10ZXN0LXBhc3N3b3Jk" #test_user credentials
+        Content-Type: application/json
+      indices.create:
+        index: sparse_vector_pruning_test
+        body:
+          mappings:
+            properties:
+              text:
+                type: text
+              ml.tokens:
+                type: sparse_vector
+                index_options:
+                  prune: true
+                  pruning_config:
+                    tokens_freq_ratio_threshold: 5.0
+                    tokens_weight_threshold: 3.5
+              embeddings:
+                type: sparse_vector
+
+  - match: { status: 400 }
+  - match: { error.type: "mapper_parsing_exception" }
+  - match: { error.reason: "Failed to parse mapping: [pruning_config] field [tokens_weight_threshold] field should be a number between 0.0 and 1.0" }
+
+---
+"Check sparse_vector token pruning index_options in query":
```

Comment:
Could we please also add tests that override the default behavior? 

For example, explicitly sending in a pruning config on queries where pruning is disabled in the mapping, and also explicitly sending in prune:false where pruning is set in the mapping?